data/mods/gen7/random-teams.ts

```diff
@@ -952,6 +952,18 @@ export class RandomGen7Teams extends RandomTeams {
 			(species.randomDoubleBattleMoves || species.randomBattleMoves) :
 			species.randomBattleMoves;
 		const movePool = (randMoves || Object.keys(Dex.getLearnsetData(species.id).learnset!)).slice();
+		if (this.format.gameType === 'multi') {
+			// Random Multi Battle uses doubles move pools, but Ally Switch fails in multi battles
+			const allySwitch = movePool.indexOf('allyswitch');
+			if (allySwitch !== undefined) {
```

Comment:
This doesn't work. [`Array#indexOf`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/indexOf) never returns `undefined`; if the value is not in the array, it returns -1. I fixed this in 2baca9339d52e9523dd0dd9f9105eb9309743ff0.